### PR TITLE
Update BlockBandedMatrix iteration for new release

### PR DESCRIPTION
**Pull request by ChrisRackauckas**
@dlfivefifty what's the right translation for `BlockArrays.axes(BlockBandedMatrices.cumulsizes(Jac,1))`?

### CODE REVIEW

**Comment by dlfivefifty** [comment]
I think just ` blockaxes(Jac,1) `. Though note that returns a ` Block ` - valued range

File: src/iteration_utils.jl [hunk 1]
```diff
@@ -22,84 +22,85 @@ _use_findstructralnz(::SparseMatrixCSC) = false
...
-                                unsafe_store!(p, b_v[k], (j-1)*st + μ + k - j + 1)
+            @inline function _colorediteration!(Jac::BlockBandedMatrices.BandedBlockBandedMatrix,
+                                                sparsity::BlockBandedMatrices.BandedBlockBandedMatrix,
+                                                rows_index,cols_index,vfx,colorvec,color_i,ncols)
+                λ,μ = BlockBandedMatrices.subblockbandwidths(Jac)
+                rs = BlockArrays.blocklasts(BlockArrays.axes(Jac,1)) # column block sizes
+                cs = BlockArrays.blocklasts(BlockArrays.axes(Jac,2))
+                b = BlockBandedMatrices.BlockArray(vfx,rs)
+                c = BlockBandedMatrices.BlockArray(colorvec,cs)
+                @inbounds for J=BlockBandedMatrices.Block.(1:BlockBandedMatrices.nblocks(Jac,2))
```
Comment (dlfivefifty) [inline src/iteration_utils.jl]: Should be ` J= blockaxes(Jac,2) `

File: src/iteration_utils.jl [hunk 2]
```diff
@@ -22,84 +22,85 @@ _use_findstructralnz(::SparseMatrixCSC) = false
...
-                            @inbounds for k = 1:m
-                                V[k,j] = b_v[k]
+            @inline function _colorediteration!(Jac::BlockBandedMatrices.BlockBandedMatrix,
+                                                sparsity::BlockBandedMatrices.BlockBandedMatrix,
+                                                rows_index,cols_index,vfx,colorvec,color_i,ncols)
+                rs = BlockArrays.axes((BlockBandedMatrices.cumulsizes(Jac,1),)) # column block sizes
+                cs = BlockArrays.axes((BlockBandedMatrices.cumulsizes(Jac,2),))
+                b = BlockBandedMatrices.BlockArray(vfx,rs)
+                c = BlockBandedMatrices.BlockArray(colorvec,cs)
+                @inbounds for J=BlockBandedMatrices.Block.(1:BlockBandedMatrices.nblocks(Jac,2))
```
Comment (dlfivefifty) [inline src/iteration_utils.jl]: ` J = blockaxes(Jac,2) `

**Comment by ChrisRackauckas** [comment]
>I think just blockaxes(Jac,1). Though note that returns a Block - valued range

Yeah, the fact that it's a Block-valued range seems to be an issue. Do I just need to turn those from Block to Int?

**Comment by dlfivefifty** [comment]
` Int(::Block) ` turns a block into an ` Int `

**Comment by ChrisRackauckas** [comment]
Looks like whatever `BlockBandedMatrices.cumulsizes` was doing was needed, since it needs to be dome some kind of summation? I tried just using `cumsum(Int.(cs))` but that doesn't seem to be it.

**Comment by dlfivefifty** [comment]
` blocklasts ` does the same thing as the previous cumulsizes: it gives the last index in each block

**Comment by ChrisRackauckas** [comment]
But it seems `blocklasts` can't act on what comes out of `blockaxis`?

**Comment by dlfivefifty** [comment]
Right, you need to call something like ` blocklasts(axes(A,1)) `

**Comment by ChrisRackauckas** [comment]
```
DimensionMismatch("block size for dimension 1: [100, 200, 300, 400, 500, 600, 700, 800, 900, 1000, 1100, 1200, 1300, 1400, 1500, 1600, 1700, 1800, 1900, 2000, 2100, 2200, 2300, 2400, 2500, 2600, 2700, 2800, 2900, 3000, 3100, 3200, 3300, 3400, 3500, 3600, 3700, 3800, 3900, 4000, 4100, 4200, 4300, 4400, 4500, 4600, 4700, 4800, 4900, 5000, 5100, 5200, 5300, 5400, 5500, 5600, 5700, 5800, 5900, 6000, 6100, 6200, 6300, 6400, 6500, 6600, 
6700, 6800, 6900, 7000, 7100, 7200, 7300, 7400, 7500, 7600, 7700, 7800, 7900, 8000, 8100, 8200, 8300, 8400, 8500, 8600, 8700, 8800, 8900, 9000, 9100, 9200, 9300, 9400, 9500, 9600, 9700, 9800, 9900, 10000] does not sum to the array size: 10000")
```

It needs the opposite of cumsum or something?

File: src/iteration_utils.jl [hunk 3]
```diff
@@ -22,84 +22,85 @@ _use_findstructralnz(::SparseMatrixCSC) = false
...
-                    m,n = size(V)
-                    @inbounds for j=1:n
-                        if c_v[j] == color_i
-                            @inbounds for k=max(1,j-μ):min(m,j+λ)
-                                unsafe_store!(p, b_v[k], (j-1)*st + μ + k - j + 1)
+            @inline function _colorediteration!(Jac::BlockBandedMatrices.BandedBlockBandedMatrix,
+                                                sparsity::BlockBandedMatrices.BandedBlockBandedMatrix,
+                                                rows_index,cols_index,vfx,colorvec,color_i,ncols)
+                λ,μ = BlockBandedMatrices.subblockbandwidths(Jac)
+                rs = BlockArrays.blocklasts(BlockArrays.axes(Jac,1)) # column block sizes
```
Comment (dlfivefifty) [inline src/iteration_utils.jl]: Probably should be ` blocklengths(axes(Jac,1)) ` to get the lengths of the blocks

Comment (ChrisRackauckas) [reply src/iteration_utils.jl]: works

**Comment by ChrisRackauckas** [comment]
This is likely the last one. I can't figure out what's supposed to replace the `BlockBandedMatrices.blocksize(Jac,(blockcolrange[1].n[1],J.n[1]))` call. From the API docs it looks like there's only `BlockBandedMatrices.blocksize(Jac)` dispatches now? I tried to see if I could find out where it went in https://github.com/JuliaArrays/BlockArrays.jl/commit/3addf8fdbbe4b9950372942b305a73609c80fb98 but couldn't find it.

**Comment by dlfivefifty** [comment]
I think:
```Julia 
length.(getindex.(axes(Jac), (blockcolrange[1], J)))
```
I could add a convenience function for this, but to be honest the old names in BlockArrays.jl we’re always confusing to me, so my preference is to be explicit instead of adding more to the interface.

**Pull request merged**